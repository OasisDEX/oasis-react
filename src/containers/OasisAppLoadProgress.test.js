/* global shallow describe it expect */
/* eslint-disable import/first */
import React from 'react';
import { fromJS } from 'immutable';

import {
  OasisAppLoadProgressWrapper,
  mapStateToProps,
  mapDispatchToProps
} from './OasisAppLoadProgress';
import { shallow } from 'enzyme';

describe('(Container) OasisAppLoadProgress', () => {
  const state = fromJS(global.storeMock);
  const initialProps = mapStateToProps(state);
  const initialActions = mapDispatchToProps(x => x);
  const props = {
    ...initialActions,
    ...initialProps
  };

  it('will receive right props', () => {
    expect(initialProps).toMatchSnapshot();
  });


  it('will receive right actions', () => {
    expect(initialActions).toMatchSnapshot();
  });

  it('should render', () => {
    const wrapper = shallow(
      <OasisAppLoadProgressWrapper {...props}/>
    );
    expect(wrapper).toMatchSnapshot();
  });

});
