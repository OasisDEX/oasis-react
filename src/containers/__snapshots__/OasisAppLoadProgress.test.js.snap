// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Container) OasisAppLoadProgress should render 1`] = `
<OasisStatus
  activeTradingPairOffersInitiallyLoaded={false}
  initialMarketHistoryLoaded={true}
  loadProgress={66.66666666666666}
  name="kovan"
  offersLoadProgress="NaN"
  status="NETWORK/ONLINE"
/>
`;

exports[`(Container) OasisAppLoadProgress will receive right actions 1`] = `
Object {
  "actions": Object {},
}
`;

exports[`(Container) OasisAppLoadProgress will receive right props 1`] = `
Object {
  "activeTradingPairOffersInitiallyLoaded": false,
  "initialMarketHistoryLoaded": true,
  "loadProgress": 66.66666666666666,
  "networkName": "kovan",
  "networkStatus": "NETWORK/ONLINE",
  "offersLoadProgress": "NaN",
}
`;
